pkg/gui/context.go

```diff
@@ -6,20 +6,21 @@ import (
 	"sort"
 	"strings"
 
+	"github.com/jesseduffield/generics/maps"
+	"github.com/jesseduffield/generics/slices"
 	"github.com/jesseduffield/gocui"
 	"github.com/jesseduffield/lazygit/pkg/gui/context"
 	"github.com/jesseduffield/lazygit/pkg/gui/types"
 )
 
 func (gui *Gui) popupViewNames() []string {
-	result := []string{}
-	for _, context := range gui.State.Contexts.Flatten() {
-		if context.GetKind() == types.PERSISTENT_POPUP || context.GetKind() == types.TEMPORARY_POPUP {
-			result = append(result, context.GetViewName())
-		}
-	}
+	popups := slices.Filter(gui.State.Contexts.Flatten(), func(c types.Context) bool {
+		return c.GetKind() == types.PERSISTENT_POPUP || c.GetKind() == types.TEMPORARY_POPUP
+	})
 
-	return result
+	return slices.Map(popups, func(c types.Context) string {
+		return c.GetViewName()
+	})
```

Comment:
I wish I could use Filter and Map as methods so that I could chain them together here without needing to introduce the intermediary variable, but it's not possible to have a generic Map method on a List struct.